module/VuFind/src/VuFind/OAuth2/Repository/AbstractTokenRepository.php

```diff
@@ -99,7 +99,7 @@ public function persistNew($token)
         }
         $userIdentifierField = $this->oauth2Config['Server']['userIdentifierField'] ?? 'id';
         $user = $this->userService->getUserByField($userIdentifierField, $userIdentifier);
-        $row->setUser($user);
+        $row->setUser($userIdentifier);
```

Comment:
This needs to be reverted -- setUser should accept a user object. Also, this revised version of the logic will only work when $userIdentifierField is set to 'id', which is not guaranteed. If we look up users by a different field, then it will break.